File: ocs-client/src/app/student-preference/Preference.ts
export class Preference {

    private preferenceId:Number;	
	private preferenceNumber:Number;	
	private instituteCourse;
    private formNumber:Number;

    getPreferenceId()
    {
        return this.preferenceId;
    }

    setPreferenceId(preferenceId:Number)
    {
        this.preferenceId=preferenceId;
    }

    getFormNumber()
    {
        return this.formNumber;
    }

    setFormNumber(formNumber:Number)
    {
        this.formNumber=formNumber;
    }

    getInstituteCourse()
    {
        return this.instituteCourse;
    }

    setInstituteCourse(instituteCourse)
    {
        this.instituteCourse=instituteCourse;
    }
    
    
    setPreferenceNumber(preferenceNumber:Number)
    {
        this.preferenceNumber=preferenceNumber;
    }
    getPreferenceNumber()
    {
        return this.preferenceNumber;
    }
    
    constructor(preferenceId:Number,instituteCourse,preferenceNumber:Number,formNumber:Number) 
    {
        this.formNumber=formNumber;
        this.preferenceId=preferenceId;
        this.preferenceNumber=preferenceNumber;
        this.instituteCourse=instituteCourse;
    }
}

File: ocs-server/src/org/sunbeam/ocs/entities/Preference.java
package org.sunbeam.ocs.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
public class Preference {

	private long preferenceId;			//PREFERENCE_ID IN DATABASE
	private int preferenceNumber;		//PREFERENCE_NUMBER IN DATABASE
	private int roundNumber;			//ROUND_NUMBER IN DATABASE
	

	private InstituteCourse instituteCourse;
	private StudentAccount studentAccount;
	
	public Preference() {
		
	}


	public Preference(long preferenceId , InstituteCourse instituteCourse, int preferenceNumber,StudentAccount studentAccount,
			int roundNumber) {
		super();
		this.preferenceId = preferenceId;
		this.instituteCourse = instituteCourse;
		this.preferenceNumber = preferenceNumber;
		this.studentAccount = studentAccount;
	}
	
	//getters and setters

	@Id
	@Column(name="PREFERENCE_ID")
	public long getPreferenceId() {
		return preferenceId;
	}


	public void setPreferenceId(long preferenceId) {
		this.preferenceId = preferenceId;
	}

	@ManyToOne
	@JoinColumn(name="INSTITUTE_COURSE_ID")
	public InstituteCourse getInstituteCourse() {
		return instituteCourse;
	}


	public void setInstituteCourse(InstituteCourse instituteCourse) {
		this.instituteCourse = instituteCourse;
	}


	@ManyToOne
	@JoinColumn(name="FORM_NUMBER")
	public StudentAccount getStudentAccount() {
		return studentAccount;
	}


	public void setStudentAccount(StudentAccount studentAccount) {
		this.studentAccount = studentAccount;
	}



	@Column(name="PREFERENCE_NUMBER")
	public int getPreferenceNumber() {
		return preferenceNumber;
	}


	public void setPreferenceNumber(int preferenceNumber) {
		this.preferenceNumber = preferenceNumber;
	}

	


	@Column(name="ROUND_NUMBER")
	public int getRoundNumber() {
		return roundNumber;
	}


	public void setRoundNumber(int roundNumber) {
		this.roundNumber = roundNumber;
	}	

}
